File: Sum of two values/solve.py
import sys
input = sys.stdin.readline

def run_tets():
    n,target = map(int,input().split())
    nums = list(map(int,input().split()))

    vec = [[nums[i],i+1] for i in range(n)]

    vec.sort()
    l = 0
    r = n - 1
    while l < r:
        value = vec[l][0] + vec[r][0]
        if value == target:
            print(vec[l][1],vec[r][1])
            return
        elif value > target:
            r -= 1
        else:
            l +=1
    print('IMPOSSIBLE')
        

def main():
    
    # number of test cases
    run_tets()

if __name__ == "__main__":
    main()

File: Sum of two values/solve.cpp
#include<iostream>
#include<map>
using namespace std;

int main(){
    int n,x;
    cin>>n>>x;
    map<int,int> m;
    for(int i = 1;i <= n; ++i)
    {
        int a;
        cin>>a;
        int y = x-a;
        if (m[y]){
            cout<<m[y]<<" "<<i<<"\n";
            return 0;
        }
        m[a] = i;
    }
    cout<<"IMPOSSIBLE\n";
}